### PR TITLE
Display gl_get_attribute unit test

### PR DESCRIPTION
**Pull request by jtoloff**
Please let me know if there is any functionality I failed to test or if my existing tests are insufficient. I am a first-time contributor, and I am looking to improve my skills.

### CODE REVIEW

**Comment by MyreMylar** [comment]
Looks like you could also do a separate test that checks the SDL2 only values that is skipped under SDL1? There should be a few examples in the tests of how to skip a test under SDL1.

SDL2 only values according to the docs:

```
GL_ACCELERATED_VISUAL,
GL_CONTEXT_MAJOR_VERSION, GL_CONTEXT_MINOR_VERSION,
GL_CONTEXT_FLAGS, GL_CONTEXT_PROFILE_MASK,
GL_SHARE_WITH_CURRENT_CONTEXT,
GL_CONTEXT_RELEASE_BEHAVIOR,
GL_FRAMEBUFFER_SRGB_CAPABLE
```

example test skipped under SDL1:

https://github.com/pygame/pygame/blob/48571a6b060b5c528ba584684e9120cec96afa99/test/rect_test.py#L848-L864

**Comment by MyreMylar** [comment]
Also - you'll have to skip both tests when the video driver is 'dummy' like it is on Travis and Appveyor. See:

https://github.com/pygame/pygame/blob/48571a6b060b5c528ba584684e9120cec96afa99/test/display_test.py#L317-L329

**Comment by jtoloff** [comment]
I believe I have correctly made all the requested changes in my most recent commit. Instead of making a new test for SDL1 and SDL2, I added conditional check within the original test function that checks the SDL version, much like another unit test in display_test.py:

https://github.com/pygame/pygame/blob/48571a6b060b5c528ba584684e9120cec96afa99/test/display_test.py#L321-L329

Please let me know if there are any other changes needed. Thank you for your help thus far!

File: test/display_test.py [hunk 1]
```diff
@@ -200,9 +206,65 @@ def todo_test_gl_get_attribute(self):
...
+        pygame.display.gl_set_attribute(pygame.GL_MULTISAMPLESAMPLES,0)
+        pygame.display.gl_set_attribute(pygame.GL_STEREO,0)
+
+        # assign SDL2-supported values with gl_set_attribute (if applicable)
+        if(SDL2):
+            pygame.display.gl_set_attribute(pygame.GL_ACCELERATED_VISUAL,0)
+            pygame.display.gl_set_attribute(pygame.GL_CONTEXT_MAJOR_VERSION,1)
+            pygame.display.gl_set_attribute(pygame.GL_CONTEXT_MINOR_VERSION,1)
+            pygame.display.gl_set_attribute(pygame.GL_CONTEXT_FLAGS, 0)
+            pygame.display.gl_set_attrubute(pygame.GL_CONTEXT_PROFILE_MASK,0)
```
Comment (MyreMylar) [inline test/display_test.py:228]: spelling mistake here, should be:

`pygame.display.gl_set_attribute(pygame.GL_CONTEXT_PROFILE_MASK,0)`

File: test/display_test.py [hunk 2]
```diff
@@ -200,9 +206,65 @@ def todo_test_gl_get_attribute(self):
...
+            pygame.display.gl_set_attribute(pygame.GL_CONTEXT_RELEASE_BEHAVIOR,1)
+            pygame.display.gl_set_attribute(pygame.GL_FRAMEBUFFER_SRGB_CAPABLE,0)
+
+        # call set_mode with OPENGL flag
+        screen = display.set_mode((0,0), pygame.OPENGL)
+
+        # test SDL1 flags using gl_get_attribute
+        self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_ALPHA_SIZE),8)
+        self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_DEPTH_SIZE),32)
+        self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_STENCIL_SIZE),0)
```
Comment (MyreMylar) [inline test/display_test.py:239]: Test fails when run locally on windows here:

```
======================================================================
FAIL: test_gl_get_attribute (__main__.DisplayModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:/Users/Dan/Programming/pygame/test/display_test.py", line 239, in test_gl_get_attribute
    self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_STENCIL_SIZE), 0)
AssertionError: 8 != 0

----------------------------------------------------------------------
```

Perhaps this attribute can't be set to 0 ? Try 8.

File: test/display_test.py [hunk 3]
```diff
@@ -200,9 +206,65 @@ def todo_test_gl_get_attribute(self):
...
+        self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_ACCUM_RED_SIZE),16)
+        self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_ACCUM_GREEN_SIZE),16)
+        self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_ACCUM_BLUE_SIZE),16)
+        self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_ACCUM_ALPHA_SIZE),16)
+        self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_MULTISAMPLEBUFFERS),0)
```
Comment (MyreMylar) [inline test/display_test.py:244]: Fails on this line (244) run locally on windows:
```
======================================================================
ERROR: test_gl_get_attribute (__main__.DisplayModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:/Users/Dan/Programming/pygame/test/display_test.py", line 244, in test_gl_get_attribute
    self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_MULTISAMPLEBUFFERS), 0)
pygame.error: OpenGL error: GL_INVALID_ENUM

======================================================================
```
I'm guessing 0 isn't a valid value in the allowed set of enums.

File: test/display_test.py [hunk 4]
```diff
@@ -200,9 +206,65 @@ def todo_test_gl_get_attribute(self):
...
+        self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_MULTISAMPLESAMPLES),0)
```
Comment (MyreMylar) [inline test/display_test.py:245]: This line also fails with a similar error:

```
======================================================================
ERROR: test_gl_get_attribute (__main__.DisplayModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:/Users/Dan/Programming/pygame/test/display_test.py", line 245, in test_gl_get_attribute
    self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_MULTISAMPLESAMPLES), 0)
pygame.error: OpenGL error: GL_INVALID_ENUM

======================================================================
```

File: test/display_test.py [hunk 5]
```diff
@@ -200,9 +206,65 @@ def todo_test_gl_get_attribute(self):
...
+        
+        # test SDL2 flags using sl_get_attribute (if applicable)
+        if(SDL2):
+            self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_ACCELERATED_VISUAL),0)
+            self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_CONTEXT_MAJOR_VERSION),1)
+            self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_CONTEXT_MINOR_VERSION),1)
+            self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_CONTEXT_FLAGS),0)
+            self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_CONTEXT_PROFILE_MASK),0)
+            self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_SHARE_WITH_CURRENT_CONTEXT),0)
+            self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_CONTEXT_RELEASE_BEHAVIOR),1)
```
Comment (MyreMylar) [inline test/display_test.py:256]: A similar error to the two above.

```
======================================================================
ERROR: test_gl_get_attribute (__main__.DisplayModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:/Users/Dan/Programming/pygame/test/display_test.py", line 257, in test_gl_get_attribute
    self.assertEqual(pygame.display.gl_get_attribute(pygame.GL_CONTEXT_RELEASE_BEHAVIOR), 1)
pygame.error: OpenGL error: GL_INVALID_ENUM

======================================================================
```

File: test/display_test.py (hunk 3 above)
Comment (MyreMylar) [reply test/display_test.py:244]: > GL_INVALID_ENUM, 0x0500
Given when an enumeration parameter is not a legal enumeration for that function. This is given only for local problems; if the spec allows the enumeration in certain circumstances, where other parameters or state dictate those circumstances, then GL_INVALID_OPERATION is the result instead.

https://www.khronos.org/opengl/wiki/OpenGL_Error

**Comment by MyreMylar** [comment]
Looks like no progress on this one. Gonna just close it up, no hard feelings, just useful to keep a clean-ish list of open PRs. You can always re-open it later if you have new inspiration.

**Pull request closed without merging**